alpha/nodejs-chart/values.schema.json

```diff
@@ -0,0 +1,109 @@
+{
+    "type": "object",
+    "required": [
+      "deploy",
+      "build",
+      "service",
+      "ingress"
+    ],
+    "additionalProperties": true,
+    "properties": {
+      "deploy": {
+        "type": "object",
+        "title": "Deploy",
+        "description": "Specifiy deployment settings",
+        "required": [
+          "replicaCount",
+          "uri",
+          "ref"
+        ],
+        "replicaCount": {
+          "default": 1,
+          "description": "Define number of nodejs pods running",
+          "title": "Replica Count",
+          "type": "integer"
+        },
+        "serviceType": {
+            "type": "string",
+            "title": "Service Type",
+            "description": "Select Service type",
+            "default": "ClusterIP"
+          },
+        "ports": {
+          "type": "object",
+          "title": "ports",
+          "description": "Specify Deployment Ports for Container",
+          "required": [
+              "name",
+              "port",
+              "targetPort",
+              "protocol"
+          ],
+          "name": {
+              "type": "string",
+              "title": "Name",
+              "default": "http"
+          },
+          "port": {
+            "type": "integer",
+            "title": "Port",
+            "description": "Define Port",
+            "default": 8080
+          },
+          "targetPort": {
+            "type": "integer",
+            "title": "Target Port",
+            "description": "Define Target Port",
+            "default": 8080
+          },
+          "protocol": {
+              "type": "string",
+              "description": "Specify which protocol to use",
+              "title": "Protocol",
+              "default": "TCP"
+          }
+        }
+      },
+      "build": {
+        "type": "object",
+        "title": "Build",
+        "description": "Specify what repo to build from",
+        "required": [
+          "enabled",
+          "uri",
+          "ref"
+        ],
+        "additionalProperties": true,
+        "properties": {
+          "enabled": {
```

Comment:
I wondered about being able to enable/disable without also including the image.name and image.tag. But I guess in both cases the container deployed comes from an ImageStream.  I assume the user would need to deploy at least once with enabled set to true and then could re-deploy without rebuilding. I'm also assuming that in the case where they want to deploy from an external docker image is not related and they'd need to drop into the yaml editing anyway. As long as my assumptions are correct them I think it's fine.

Reply:
The image.name and image.tag, with build disabled, would refer to a DockerImage (external registry)

Reply:
Or an ImageStream. Common use case would be DockerImage though, I assume.

Reply:
@deweya, if the common use case is to use with DockerImage and you have to drop into the YAML to do that would it make sense to either remove the option to disable build, or alternative add the ability to specify the image/tag (maybe only if build is disabled)? Otherwise its seems like we are potentially leading the user down a path that does not make sense?

Reply:
By dropping into the YAML, do you mean modifying the actual OCP resources? Or do you mean modifying the values?

If you mean the former, @sabre1041 gave a cool demo a while ago where he deployed an external docker image using the Quarkus chart (configured similarly as this one). He set `image.name` and `image.tag` to match the image he wanted to deploy. These values just get rendered in the Deployment's `image` field, so it's easy to deploy an external image going that route.

Alternatively, if the user doesn't have an external image and wants to build one instead, they can use the chart's build feature to produce an ImageStream or DockerImage, whose name is also determined by `image.name` and `image.tag`.

Reply:
Actually, had to refresh my memory, `image.name` actually defaults to the Helm release name, so most of the time when users want to build, they don't even need to worry about that value assuming they named their Helm release the same name they want their image to have.

Reply:
My comments were around the UI, the alpha/nodejs-chart/values.schema.json defines the elements that the user can configure through the form. When I say dropping down you YAML I meant having to switch from the form view to the YAML editor to directly edit the YAML instead of using the form.  In terms of what you mentioned, the form does not currently expose  "image.name and image.tag" which was the reason form my original comment, if you want to edit those you have to switch the YAML editor.  My question was then if you have to switch to the YAML editor for the common use case, does it make sense to expose build/no build in the form.